app/models/concerns/node_properties.rb

```diff
@@ -124,4 +124,23 @@ def raw_properties=(value)
   rescue JSON::ParserError => exception
     @raw_properties = value
   end
+
+  def merge_properties(source_properties)
```

Comment:
If we are going down this route (see next comment), maybe add one of those ugly comment/separator things to split this method from the `raw properties` block.